cmd/searcher/internal/search/search_regex.go

```diff
@@ -193,6 +190,48 @@ func regexSearch(
 	return err
 }
 
+// fileLoader loads files from the zipfile. It keeps a reference to the last
+// file it loaded, in case it's requested twice (for example first from language
+// detection, then pattern matching).
+type fileLoader struct {
+	zf              *zipFile
+	isCaseSensitive bool
+
+	// fileMatchBuf is what we run match on, fileBuf is the original data
+	// (used for the content preview).
+	fileBuf      []byte
+	fileMatchBuf []byte
```

Comment:
```suggestion
    // fullContentsBuf is the original data for the file, and is used for the content preview
	fullContentsBuf      []byte
	// matchBuf represents a (potentially case converted) sub-slice of fullContentsBuf
	// used for matching content or languages
	matchBuf []byte
```

Minor suggestion for renaming to reduce similarity between the field names.

(That said, I'm not entirely sure where the 2MB limit is being applied; is `fullContentsBuf` guaranteed to be less than 2MB?)

Reply:
Thanks for the suggestion here -- I improved the comments. For names, I'm actually going to keep the original since they better match what the variables represent: `fileMatchBuf` is not a subslice of `fileBuf`, it is either exactly the same, or a lower-cased version of it.